File: algorithms/go/sieve.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscan(reader, &n)

	start := time.Now()

	isComposite := make([]bool, n+1)
	isComposite[0] = true
	if n > 0 { isComposite[1] = true }

	for i := 2; i*i <= n; i++ {
		if !isComposite[i] {
			for j := i * i; j <= n; j += i {
				isComposite[j] = true
			}
		}
	}

	var count, last int64
	for k := 2; k <= n; k++ {
		if !isComposite[k] {
			count++
			last = int64(k)
		}
	}

	ms := float64(time.Since(start).Nanoseconds()) / 1e6

	fmt.Fprintln(writer, count)
	fmt.Fprintln(writer, last)
	fmt.Fprintf(os.Stderr, "%.3f\n", ms)
}

File: algorithms/c/sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

int main() {
    long long n;
    scanf("%lld", &n);

    // bitset برای کاهش مصرف RAM
    unsigned char *is_composite = calloc((n + 1), sizeof(unsigned char));

    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);

    is_composite[0] = is_composite[1] = 1;
    for (long long i = 2; i * i <= n; i++) {
        if (!is_composite[i]) {
            for (long long j = i * i; j <= n; j += i)
                is_composite[j] = 1;
        }
    }

    long long count = 0, last = -1;
    for (long long i = 2; i <= n; i++) {
        if (!is_composite[i]) { count++; last = i; }
    }

    clock_gettime(CLOCK_MONOTONIC, &end);
    double ms = (end.tv_sec  - start.tv_sec)  * 1000.0
              + (end.tv_nsec - start.tv_nsec) / 1e6;

    printf("%lld\n%lld\n", count, last);
    fprintf(stderr, "%.3f\n", ms);

    free(is_composite);
    return 0;
}